client/src/utils/updateMyEvents.ts

```diff
@@ -0,0 +1,24 @@
+import type { Rsvp } from "../types/Rsvp";
+import type { MyEvents } from "../types/MyEvents";
+
+export const updateMyEvents = (rsvp: Rsvp) => {
+  const rsvpDetails = { id: rsvp.id, token: rsvp.token };
+  const myEvents = localStorage.getItem("my_events");
+
+  if (myEvents) {
+    const parsedEvents = JSON.parse(myEvents) as MyEvents;
+
+    if (parsedEvents[rsvp.event_id]) {
+      parsedEvents[rsvp.event_id].push(rsvpDetails);
+    } else {
+      parsedEvents[rsvp.event_id] = [rsvpDetails];
+    }
```

Comment:
I think this might work just for maximum code golf haha (also feel free to ignore, just wanted to try it, it's probably less readable):


```suggestion
    parsedEvents[rsvp.event_id] = (parsedEvents[rsvp.event_id] || []).push(rsvpDetails);
```

Reply:
(and we could probably do something similar for the "root" structure as well)

```ts
const myEvents = JSON.parse(localStorage.getItem("my_events")) || {} as MyEvents;
```